core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -1854,6 +1854,23 @@ class SocketServerTest {
     })
   }
 
+  @Test
+  def testListenBacklogSize(): Unit = {
+    val backlogSize = 128
+    props.put("socket.listen.backlog.size", backlogSize.toString)
+
+    // TCP listen backlog size is the max count of pending connections (i.e. connections such that
+    // 3-way handshake is done at kernel level and waiting to be accepted by the server application.
+    // From client perspective, such connections should be visible as already "connected")
+    // Hence, we can check if listen backlog size is properly configured by trying to connect the server
+    // without starting acceptor thread.
+    withTestableServer(KafkaConfig.fromProps(props), { testableServer =>
+      1 to backlogSize foreach { _ =>
+        assertTrue(connect(testableServer).isConnected)
```

Comment:
Naive question: Is `isConnected` immediately true or is there a risk that it is not true yet when the assertion is ran?

Reply:
As far as I checked, `isConnected` should be true here immediately.
- connect() =>
- https://github.com/adoptium/jdk11u/blob/jdk-11.0.13%2B8/src/java.base/share/classes/java/net/Socket.java#L304 =>
- https://github.com/adoptium/jdk11u/blob/jdk-11.0.13%2B8/src/java.base/share/classes/java/net/Socket.java#L454 =>
- https://github.com/adoptium/jdk11u/blob/jdk-11.0.13%2B8/src/java.base/share/classes/java/net/Socket.java#L617